app/routes.py

```diff
@@ -1,2 +1,235 @@
+
+        return make_response({
+            "goal":{
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }
+        }, 201)
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def handle_one_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+    if request.method == "GET":
+        return ({"goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }}, 200)        
+    elif request.method == "PUT":
+        form_data = request.get_json() 
+        goal.title = form_data["title"]
+        db.session.commit()
+        return make_response({"goal":{
+            "id": goal.goal_id,
+            "title": goal.title
+        }}, 200)
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'}), 200
+
+@goals_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_goals():
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        goals_response.append({
+            "id": goal.goal_id,
+            "title": goal.title
+        })
+
+    return make_response(jsonify(goals_response), 200)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET", "POST"], strict_slashes=False)
+def tasks_by_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    request_body = request.get_json()
+
+    if not goal:
+        return make_response("", 404)
+    
+    elif request.method == "POST":
+        for task_id in request_body["task_ids"]:
+            task = Task.query.get(task_id)   
+            task.goal_identity = goal_id
+        db.session.commit()
+
+        return make_response({"id": goal.goal_id,
+                "task_ids": request_body["task_ids"]
+        }, 200) 
+
+    elif request.method == "GET":
+        tasks = goal.tasks
+        if tasks == []:
+            response = {
+                "id": goal.goal_id,
+                "title": goal.title,
+                "tasks": []
+            }
+            return response
+
+        else:
+            tasks_response = []
+            for task in tasks:
+                if task.goal_identity == goal.goal_id:
+                    tasks_response.append(
+                            {
+                                "id": task.id,
+                                "goal_id": task.goal_identity,
+                                "title": task.title,
+                                "description": task.description,
+                                "is_complete": bool(task.completed_at)
+                        })
+                response = {
+                        "id": goal.goal_id,
+                        "title": goal.title,
+                        "tasks": tasks_response
+                        }
+            return make_response(jsonify(response), 200)
```

Comment:
here we could minimize some code here by defining the response at the top, then add an empty list of tasks or the tasks themselves rather than repeating lines 211-15 and 230-234